File: frontend/src/modules/graph/application/use-cases/readCommits.ts
import { CommitReader } from "@modules/graph/application/contracts/CommitReader";
import { Commit } from "@modules/graph/domain/commit";

export async function readCommits(
  commitReader: CommitReader,
  repositoryPath: string
): Promise<Commit[]> {
  return commitReader.readCommits(repositoryPath);
}

File: backend/src/app/commits/read_commits.rs
use std::path::Path;

use crate::app::commits::contracts::CommitReader;
use crate::domain::commit::Commit;
use crate::domain::errors::AppError;

pub fn read_commits(
    commit_reader: &dyn CommitReader,
    repository_path: &Path,
) -> Result<Vec<Commit>, AppError> {
    commit_reader.read_commits(repository_path)
}
